ui/components/MesherySettings.js

```diff
@@ -134,205 +134,147 @@ TabContainer.propTypes = { children : PropTypes.node.isRequired, };
 
 
 //TODO: Tabs are hardcoded everywhere
-class MesherySettings extends React.Component {
-  constructor(props) {
-    super(props);
-    const {
-      k8sconfig, meshAdapters, grafana, prometheus, router : { asPath }
-    } = props;
-
-    this._isMounted = false;
-
-    //TODO: Extract this logic and clean it
-    let tabVal = 0, subTabVal = 0;
-    const splittedPath = asPath.split('#');
-    if (splittedPath.length >= 2 && splittedPath[1]) {
-      const subTabPath = splittedPath[1].split('/');
-
-      switch (subTabPath[0]) {
-
-        case 'environment':
-          tabVal = 0;
-          break;
-        case 'service-mesh':
-          tabVal = 1;
-          break;
-        case 'metrics':
-          tabVal = 2;
-          break;
-        case 'system':
-          tabVal = 3;
-          break
-        case 'meshmodel-summary' :
-          tabVal = 4;
-          break
-        // case 'performance':
-        //   tabVal = 3;
-        //   break;
-      }
-      if (subTabPath.length >= 2 && subTabPath[1]) {
-        switch (subTabPath[1]) {
-          case 'inclusterconfig':
-            subTabVal = 0;
-            break;
-          case 'outclusterconfig':
-            subTabVal = 1;
-            break;
-          case 'grafana':
-            subTabVal = 0;
-            break;
-          case 'prometheus':
-            subTabVal = 1;
-            break;
-        }
-      }
-    }
-    this.state = {
-      completed : {},
-      k8sconfig,
-      meshAdapters,
-      grafana,
-      prometheus,
-      tabVal,
-      subTabVal,
-      modelsCount : 0,
-      componentsCount : 0,
-      relationshipsCount : 0,
-      isMeshConfigured : k8sconfig.clusterConfigured,
-
-      // Array of scanned prometheus urls
-      scannedPrometheus : [],
-      // Array of scanned grafan urls
-      scannedGrafana : [],
-    };
-
-    this.systemResetPromptRef = React.createRef();
-  }
+const MesherySettings = (props) => {
+  const {
+    k8sconfig,
+    meshAdapters,
+    grafana,
+    prometheus,
+    classes,
+  } = props;
+
+  const  notify  = useNotification();
+  const [modelsCount, setModelsCount] = useState(0);
+  const [componentsCount, setComponentsCount] = useState(0);
+  const [relationshipsCount, setRelationshipsCount] = useState(0);
+  const [scannedGrafana, setScannedGrafana] = useState([]);
+  const [scannedPrometheus, setScannedPrometheus] = useState([]);
+  const [tabVal, setTabVal] = useState(0);
+  const [subTabVal, setSubTabVal] = useState(0);
+  const [isMeshConfigured] = useState(k8sconfig.clusterConfigured);
+  const systemResetPromptRef = useRef(null);
+
+
+  useEffect(() => {
+    if (
+      JSON.stringify(props.k8sconfig) !== JSON.stringify(k8sconfig) ||
+       JSON.stringify(props.meshAdapters) !== JSON.stringify(meshAdapters)
+    ) {
+
+      k8sconfig (props.k8sconfig)
```

Comment:
what does this do , isnt k8sconfig destructured from props

Reply:
@aabidsofi19 
Actually here we are checking the props.k8sconfig state conditionally that if this condition {(JSON.stringify(props.k8sconfig) !== JSON.stringify(k8sconfig)} is true then props.k8config has changed . and if it's changed then we are updating the k8sconfig local variable. this is the same logic that are mention into class based component . And same for next line